File: archived/2022-11/ton3_b.py
# import math, heapq, bisect, itertools, functools
# from collections import deque, Counter, defaultdict, OrderedDict

def foo(n, s):
    ans = len([i for i in s if i == '0']) * len([i for i in s if i == '1'])
    count = 0
    char = '/'
    for i in s:
        if i == char:
            count += 1
        else:
            ans = max(ans, count**2)
            char = i
            count = 1
    ans = max(ans, count**2)
    return ans

# python sample.py < input.txt

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input()
        print(foo(n, s))

File: archived/2022-11/ton3_b.cpp
#include <bits/stdc++.h>
using namespace std;

long long foo(long long n, string& s) {
    long long zeros = 0;
    long long ones = 0;
    for (auto &i:s) {
        if (i == '0') zeros++;
        else ones++;
    }
    long long ans = zeros * ones;
    long long count = 0;
    char c = '/';
    for (auto &i:s) {
        if (c != i) {
            ans = max(ans, count*count);
            count = 1;
            c = i;
        } else count++;
    }
    ans = max(ans, count*count);
    return ans;
}

// g++ sample.cpp -o run
// ./run < input.txt

int main() {
    long long t, n;
    string s;
    cin >> t;
    for (long long i=0; i<t; i++) {
        cin >> n;
        cin >> s;
        cout << foo(n, s) << endl;
    }
}